casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -144,6 +146,48 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
       }
   }
 
+  def computeBalance(start: StateHash)(user: ByteString): F[Long] =
+    withResetRuntime(start)(computeBalance(_)(user))
+
+  private def balanceQuerySource(user: ByteString): String =
+    s"""
+       | new rl(`rho:registry:lookup`), revAddressOps(`rho:rev:address`), revVaultCh in {
+       |   rl!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh)
+       |   | for (@(_, RevVault) <- revVaultCh) {
+       |     new vaultCh, revAddressCh in {
+       |       revAddressOps!("fromPublicKey", "${Base16.encode(user.toByteArray)}".hexToBytes(), *revAddressCh)
+       |       | for(@revAddress <- revAddressCh) {
+       |         @RevVault!("findOrCreate", revAddress, *vaultCh)
+       |         | for(@vaultEither <- vaultCh){
+       |           match vaultEither {
+       |             (true, vault) => {
+       |               @vault!("balance", "__SCALA__")
+       |             }
+       |             (false, error) => {
+       |               @"__SCALA__"!(error)
+       |             }
+       |           }
+       |         }
+       |       }
+       |     }
+       |   }
+       | }
+     """.stripMargin
+
+  // TODO: Determine if accumulated values on @"__SCALA__" interfere with balance/payment processing.
+  private def computeBalance(runtime: Runtime[F])(user: ByteString): F[Long] =
+    computeEffect(runtime)(
```

Comment:
So is the call to computeBalance followed by a createCheckpoint call before we call reset - as in are we permanently changing the tuplespace trie?

Reply:
No, the balance check will not change the tuplespace trie.

Reply:
@ArturGajowy Be sure to note the above as you continue the implementation.

Reply:
@KentShikama @JosDenmark I believe the contrary should be the case. If we're interacting with the tuplespace in any way, we must leave a `trace` of that. This way we constrain the potential for the validators cheating and make our lives easier when debugging any controversial behavior on chain.